Sources/Tracing/Docs.docc/Guides/TraceYourApplication.md

```diff
@@ -0,0 +1,443 @@
+
+Specifically, it is recommended to bootstrap systems in the following order:
+
+1. [Swift Log](https://github.com/apple/swift-log#default-logger-behavior)'s `LoggingSystem`
+2. [Swift Metrics](https://github.com/apple/swift-metrics#selecting-a-metrics-backend-implementation-applications-only)' `MetricsSystem`
+3. [Swift Distributed Tracing](https://github.com/apple/swift-distributed-tracing)'s `InstrumentationSystem`
+4. Finally, any other parts of your application
+
+This is because tracing systems may attempt to emit logs or metrics about their status etc.
+
+If you intend to use trace identifiers for log correlation (i.e. logging a `trace-id` in every log statement that is part of a trace),
+then don't forget to also configure a swift-lot `MetadataProvider`.
+
+A typical bootstrap could look something like this:
+
+```swift
+import OpenTelemetry
+import StatsdMetrics
+import FancyLogging
+
+extension Logger.MetadataProvider {
+
+    // Include the following OpenTelemetry tracer specific metadata in log statements:
+    static let otel = Logger.MetadataProvider { baggage in
+        guard let spanContext = baggage?.spanContext else { return nil }
+        return [
+          "trace-id": "\(spanContext.traceID)",
+          "span-id": "\(spanContext.spanID)",
+        ]
+    }
+}
+
+// 1) bootstrap swift-log: stdout-logger
+LoggingSystem.bootstrap(
+  StreamLogHandler.standardOutput,
+  metadataProvider: .otel
+)
+
+// 2) bootstrap metrics: statsd
+let statsdClient = try StatsdClient(host: host, port: port)
+MetricsSystem.bootstrap(statsdClient)
+
+// 3) bootstrap swift-distributed-tracing: open-telemetry
+let group: MultiThreadedEventLoopGroup = ...
+let otel = OTel(serviceName: "onboarding", eventLoopGroup: group)
+
+try otel.start().wait()
+InstrumentationSystem.bootstrap(otel.tracer())
+
+// 4) Continue starting your application ...
+```
+
+#### Bootstrapping multiple instruments using MultiplexInstrument
+
+If you'd find yourself in need of using multiple instrumentation or tracer implementations you can group them in a `MultiplexInstrument` first, which you then pass along to the `bootstrap` method like this:
+
+```swift
+InstrumentationSystem.bootstrap(MultiplexInstrument([FancyInstrument(), OtherFancyInstrument()]))
+```
+
+`MultiplexInstrument` will then call out to each instrument it has been initialized with.
+
+### Introducing Trace Spans
+
+The primary way you interact with distributed tracing is by starting ``Span``s.
+
+Spans form hierarchies with their parent spans, and end up being visualized using various tools, usually in a format similar to gant charts. So for example, if we had multiple operations that compose making dinner, they would be modelled as child spans of a main `makeDinner` span. Any sub tasks are again modelled as child spans of any given operation, and so on.
+
```

Comment:
Add link to https://opentelemetry.io/docs/concepts/observability-primer/ ?

Reply:
Yeha might as well, We're not "limited to" otel but they have some good guides, I'll add a links